tiledb/sm/cpp_api/log.h

```diff
@@ -0,0 +1,59 @@
+/**
+ * @file   log.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file declares the C++ API's logger.
+ */
+
+#ifndef TILEDB_CPP_API_LOG_H
+#define TILEDB_CPP_API_LOG_H
+
+#include "context.h"
+#include "tiledb.h"
+
+namespace tiledb {
+
+namespace impl {
+
+/**
+ * Logs events from the C++ API.
+ * TODO: This is currently a stub; write a proper implementation.
+ */
+class Log {
+ public:
+  Log() = delete;
+
+  static void warn(const std::string&) noexcept {
```

Comment:
A free function called `log_warn()` would be adequate. The class definition is overhead I don't think we'll ever make use of.